src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -282,6 +282,103 @@ def add_uc_role_policy(self, role_name, policy_name, s3_prefixes: set[str], acco
             return False
         return True
 
+    def add_migration_role(self, role_name: str) -> str | None:
+        aws_role_trust_doc = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {"Effect": "Allow", "Principal": {"Service": "ec2.amazonaws.com"}, "Action": "sts:AssumeRole"}
+            ],
+        }
+        assume_role_json = self._get_json_for_cli(aws_role_trust_doc)
+        role = self._run_json_command(
+            f"iam create-role --role-name {role_name} --assume-role-policy-document {assume_role_json}"
+        )
+        if not role:
+            return None
+
+        return role["Role"]['Arn']
+
+    def get_instance_profile(self, instance_profile_name: str) -> str | None:
+        instance_profile = self._run_json_command(
+            f"iam get-instance-profile --instance-profile-name {instance_profile_name}"
+        )
+
+        if not instance_profile:
+            return None
+
+        return instance_profile["InstanceProfile"]["Arn"]
+
+    def create_instance_profile(self, instance_profile_name: str) -> str | None:
+        instance_profile = self._run_json_command(
+            f"iam create-instance-profile --instance-profile-name {instance_profile_name}"
+        )
+
+        if not instance_profile:
+            return None
+
+        return instance_profile["InstanceProfile"]["Arn"]
+
+    def add_role_to_instance_profile(self, instance_profile_name: str, role_name: str):
+        # there can only be one role associated with iam instance profile
+        self._run_command(
+            f"iam add-role-to-instance-profile --instance-profile-name {instance_profile_name} --role-name {role_name}"
+        )
+
+    def add_or_update_migration_policy(
```

Comment:
this method belongs to `AWSResourcePermissions` class, `AWSResources` should get only `put_role_policy` method.